modules/core/src/main/java/org/locationtech/jts/geom/prep/PreparedPoint.java

```diff
@@ -45,4 +45,7 @@ public boolean intersects(Geometry g)
   	 */
     return isAnyTargetComponentInTest(g);
   }  
+  protected int getSortIndex() {
+      return BasicPreparedGeometry.SORTINDEX_POINT;
```

Comment:
Can't this just delegate to the parent geometry `getSortIndex `?  Then the explicit reference to the constant is not needed.

Reply:
the problem is that `BasicPreparedGeometry` is trying to delegate to it's wrapped geometry, but it can't access the protected methods in the wrapped geometry in order to invoke them... without having `PreparedGeometry` extend `Geometry`, I think the other option would be to use reflection to access the protected method...

Reply:
Right, understood.    It's a fundamental problem with this approach.  I'm not keen on injecting this into the core JTS codebase.  If you really need this can't you create a wrapper object external to JTS?

Reply:
@dr-jts, as an FYI, I'm the one pushing for this.:)  

I'm seeing the PreparedGeometry as a good story for working with optimizations from JTS through GeoTools/GeoServer to end applications.  In GeoTools, it seems like there should be a 'prepareFilter' method/visitor would optimize spatial filters involving complex geometries.